compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1076,6 +1076,21 @@ pub struct VarDebugInfo<'tcx> {
 // BasicBlock
 
 rustc_index::newtype_index! {
+    /// The unit of the MIR [control-flow graph][CFG].
+    ///
+    /// There are no branches (e.g., `if`s, function calls, etc.) within a basic block, which makes
+    /// it easier to do [data-flow analyses] and optimizations. Instead, branches are represented
+    /// as an edge in a graph between basic blocks.
+    ///
+    /// Basic blocks consist of a series of [statements][Statement], ending with a
+    /// [terminator][Terminator]. Basic blocks can have multiple predecessors and successors.
```

Comment:
There is a specific MIR pass (`CriticalCallEdges`) that removes critical edges (edges that go from a multi-successor node to a multi-predecessor node) as some analysis require no critical edges. This seems like it would be worth mentioning here.